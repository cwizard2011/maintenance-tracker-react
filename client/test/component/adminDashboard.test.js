import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import DefaultAdminDashboard, { AdminDashboard } from '../../src/components/AdminDashboard';

const mockStore = configureMockStore([thunk]);

const props = {
  userRole: 'admin'
};
const initialState = {
  authReducer: {
    user: {
      user_role: 'admin'
    }
  }
};
const store = mockStore(initialState);

describe('Admin dashboard', () => {
  it('render admin dashboard correctly', () => {
    const wrapper = shallow(<AdminDashboard {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').length).toEqual(2);
  });
  it('render redirect non admin', () => {
    const newProps = {
      userRole: 'user',
      history: {
        push: jest.fn()
      }
    };
    const wrapper = shallow(<AdminDashboard {...newProps} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').length).toEqual(0);
  });
  it('render connected component', () => {
    const wrapper = shallow(<DefaultAdminDashboard store={store} {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
