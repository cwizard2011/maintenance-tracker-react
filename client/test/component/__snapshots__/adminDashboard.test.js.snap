// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin dashboard render admin dashboard correctly 1`] = `
<div>
  <Connect(UserNavigation) />
  <div
    className="container"
  >
    <Connect(RequestTable) />
  </div>
  <Footer />
</div>
`;

exports[`Admin dashboard render connected component 1`] = `
<AdminDashboard
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  userRole="admin"
/>
`;

exports[`Admin dashboard render redirect non admin 1`] = `""`;
